packages/firestore/src/local/query_engine.ts

```diff
@@ -121,8 +146,83 @@ export class QueryEngine {
             )
       )
       .next(result =>
-        result ? result : this.executeFullCollectionScan(transaction, query)
+        result
+          ? result
+          : this.executeFullCollectionScan(transaction, query, queryContext)
+      )
+      .next(result =>
+        result && this.indexAutoCreationEnabled
+          ? this.createCacheIndexes(
+              transaction,
+              query,
+              queryContext,
+              result.size
+            ).next(_ => result)
+          : result
+      );
+  }
+
+  createCacheIndexes(
+    transaction: PersistenceTransaction,
+    query: Query,
+    context: QueryContext,
+    resultSize: number
+  ): PersistencePromise<void> {
+    if (context.documentReadCount < this.indexAutoCreationMinCollectionSize) {
+      if (getLogLevel() <= LogLevel.DEBUG) {
```

Comment:
Technically, the `if (getLogLevel() <= LogLevel.DEBUG)` is _not_ necessary; however, it's used throughout this file to avoid the expensive call to `stringifyQuery()` if its result is not going to be used. I adopted this idiom from other places in this file so it seemed prudent to follow that pre-existing pattern.

Reply:
Got it.